### PR TITLE
[IOPAE-666] Physically delete service on service-publication

### PR DESCRIPTION
**Pull request by giamma1295**
With this pr we implement the **physical deletion of a service from the service-publication cosmosDB container** following a service-deletion action.

BTW the service remains in the cosmosDB service-lifecycle container(with status `deleted`, logical deletion), in addition the deleted service-publication entry remains replicated on the service-history container(consequently it will be possible to restore, with manual intervention, the deleted service).

#### List of Changes
- Add new Azure Function `OnServiceDeletion`, this QueueTrigger(listen on `request-deletion` queue) type function will evetually delete received services fron `service-publication` container.
- Updated `LegacyServiceWatcher` Azure Function, in order to stop creating an unpublished record when a legacy service deletion is intercepte but only a delete record for sync.
- Update `ServiceLifecycleWatcher` Azure Function in order to intercept deletetion events, both from `io-service-cms` and `legacy`, and write a message on `request-deletion` queue
- Add `delete` operation to FSMStore, both memory and cosmos
- Updated unit tests

#### Motivation and Context
Fix an edge case which is allowing a deleted service to be published.

#### How Has This Been Tested?
Unit test and Local Application Run

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: apps/io-services-cms-webapp/src/watchers/on-service-lifecycle-change.ts [hunk 1]
```diff
@@ -65,58 +65,66 @@ const onApproveHandler =
...
-    }
-  };
+  ({ item }) =>
+    pipe(
+      item,
+      onAnyChangesHandler,
+      E.right,
+      E.chain((historicizationAction) =>
+        pipe(
+          item.fsm.lastTransition !== SYNC_FROM_LEGACY ||
```
Comment (gunzip) [inline apps/io-services-cms-webapp/src/watchers/on-service-lifecycle-change.ts:113]: can you add a comment?

File: apps/io-services-cms-webapp/src/watchers/on-service-lifecycle-change.ts [hunk 2]
```diff
@@ -65,58 +65,66 @@ const onApproveHandler =
...
+            item.fsm.state === "deleted",
+          B.fold(
+            () => E.right(historicizationAction),
+            () =>
+              pipe(
+                item,
+                getSpecificAction(config),
```
Comment (gunzip) [inline apps/io-services-cms-webapp/src/watchers/on-service-lifecycle-change.ts:120]: ditto

File: packages/io-services-cms-models/src/lib/__tests__/store.cosmos.test.ts [hunk 3]
```diff
@@ -241,4 +241,54 @@ describe("store cosmos tests", () => {
       expect(result.right.version).toBe(anEtag);
     }
   });
+
+  it("shold delete the element", async () => {
```
Comment (gunzip) [inline packages/io-services-cms-models/src/lib/__tests__/store.cosmos.test.ts:245]:
```suggestion
  it("should delete the element", async () => {
```

File: packages/io-services-cms-models/src/lib/fsm/store.cosmos.ts [hunk 4]
```diff
@@ -126,5 +127,37 @@ export const createCosmosStore = <
       }))
     );
 
-  return { fetch, bulkFetch, save };
+  // https://learn.microsoft.com/en-us/rest/api/cosmos-db/delete-a-document
+  // expected status code return are:
+  // - 204: The documenti was successfully deleted.
+  // - 404: The documenti was not found.
```
Comment (gunzip) [inline packages/io-services-cms-models/src/lib/fsm/store.cosmos.ts:133]:
```suggestion
  // - 204: The document was successfully deleted.
  // - 404: The document was not found.
```

File: apps/io-services-cms-webapp/src/watchers/on-service-lifecycle-change.ts (hunk 1 above)
Comment (giamma1295) [reply apps/io-services-cms-webapp/src/watchers/on-service-lifecycle-change.ts:120]: 2d32e4c8978af5f93573a9c52ff4768d6692b4cc

File: apps/io-services-cms-webapp/src/watchers/on-service-lifecycle-change.ts (hunk 2 above)
Comment (giamma1295) [reply apps/io-services-cms-webapp/src/watchers/on-service-lifecycle-change.ts:128]: 2d32e4c8978af5f93573a9c52ff4768d6692b4cc

File: packages/io-services-cms-models/src/lib/__tests__/store.cosmos.test.ts (hunk 3 above)
Comment (giamma1295) [reply packages/io-services-cms-models/src/lib/__tests__/store.cosmos.test.ts]: d1706a746a439ca36a11a2bd465f96c14f2254f2

File: packages/io-services-cms-models/src/lib/fsm/store.cosmos.ts (hunk 4 above)
Comment (giamma1295) [reply packages/io-services-cms-models/src/lib/fsm/store.cosmos.ts]: 8e40c603ee75d47f13ca1dcb563fec36991daeda

**Review by rodoviti — APPROVED** [review]
LGTM
